ecosystem/gstreamer_plugin/gstmtltxsink.c

```diff
@@ -0,0 +1,653 @@
+#endif
+#ifndef PACKAGE
+#define PACKAGE "gst-mtl-tx-st20"
+#endif
+#ifndef PACKAGE_VERSION
+#define PACKAGE_VERSION "1.19.0.1"
+#endif
+
+enum {
+  PROP_0,
+  PROP_SILENT,
+  PROP_TX_DEV_ARGS_PORT,
+  PROP_TX_DEV_ARGS_SIP,
+  PROP_TX_DEV_ARGS_DMA_DEV,
+  PROP_TX_PORT_PORT,
+  PROP_TX_PORT_IP,
+  PROP_TX_PORT_UDP_PORT,
+  PROP_TX_PORT_PAYLOAD_TYPE,
+  PROP_TX_PORT_TX_QUEUES,
+  PROP_MAX
+};
+
+/* pad template */
+static GstStaticPadTemplate gst_mtltxsink_sink_pad_template =
+    GST_STATIC_PAD_TEMPLATE("sink", GST_PAD_SINK, GST_PAD_ALWAYS,
+                            GST_STATIC_CAPS("video/x-raw, "
+                                            "format = (string) {v210, I422_10LE},"
+                                            "width = (int) [64, 16384], "
+                                            "height = (int) [64, 8704], "
+                                            "framerate = (fraction) [0, MAX]"));
+
+#define gst_mtltxsink_parent_class parent_class
+G_DEFINE_TYPE_WITH_CODE(GstMtlTxSink, gst_mtltxsink, GST_TYPE_VIDEO_SINK,
+                        GST_DEBUG_CATEGORY_INIT(gst_mtltxsink_debug, "mtltxsink", 0,
+                                                "Mtl St2110 st20 transmition sink"));
+
+GST_ELEMENT_REGISTER_DEFINE(mtltxsink, "mtltxsink", GST_RANK_NONE, GST_TYPE_MTL_TX_SINK);
+
+static void gst_mtltxsink_set_property(GObject* object, guint prop_id,
+                                       const GValue* value, GParamSpec* pspec);
+static void gst_mtltxsink_get_property(GObject* object, guint prop_id, GValue* value,
+                                       GParamSpec* pspec);
+
+static gboolean gst_mtltxsink_sink_event(GstPad* pad, GstObject* parent, GstEvent* event);
+static GstFlowReturn gst_mtltxsink_chain(GstPad* pad, GstObject* parent, GstBuffer* buf);
+static gboolean gst_mtltxsink_query(GstPad* pad, GstObject* parent, GstQuery* query);
+
+static gboolean gst_mtltxsink_start(GstBaseSink* bsink);
+static gboolean gst_mtltxsink_stop(GstBaseSink* bsink);
+
+static gboolean gst_mtltxsink_parse_input_fmt(GstVideoInfo* info, enum st_frame_fmt* fmt);
+static gboolean gst_mtltxsink_parse_fps(GstVideoInfo* info, enum st_fps* fps);
+static struct st_frame* gst_mtltxsink_get_frame(GstMtlTxSink* sink);
+
+static gboolean gst_mtltxsink_parse_input_fmt(GstVideoInfo* info,
+                                              enum st_frame_fmt* fmt) {
+  GstVideoFormatInfo* finfo = info->finfo;
+
+  if (finfo->format == GST_VIDEO_FORMAT_v210) {
+    *fmt = ST_FRAME_FMT_V210;
+  } else if (finfo->format == GST_VIDEO_FORMAT_I420_10LE) {
+    *fmt = ST_FRAME_FMT_YUV422PLANAR10LE;
+  } else {
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+static gboolean gst_mtltxsink_parse_fps(GstVideoInfo* info, enum st_fps* fps) {
+  gint fps_div;
+  if (info->fps_n == 0 || info->fps_d == 0) {
+    return FALSE;
+  }
+
+  fps_div = info->fps_n / info->fps_d;
+
+  switch (fps_div) {
+    case 24:
```

Comment:
How about fractions, which we theoretically allow in line 113 and 159?
```suggestion
  switch (fps_div) {
    /* TODO Add fractions handling */
    case 24:
```

Reply:
added a comment in 154 
this will come with the support for rawbuffers without the src element